context.c

```diff
@@ -311,19 +312,18 @@ struct iio_context * iio_create_default_context(void)
 	char *hostname = getenv("IIOD_REMOTE");
 
 	if (hostname) {
-		struct iio_context *ctx;
+		if (strlen(hostname) > 2 && strlen(hostname) < MAXHOSTNAMELEN + 4) {
+			struct iio_context *ctx;
 
-		ctx = iio_create_context_from_uri(hostname);
-		if (ctx)
-			return ctx;
-
-#ifdef WITH_NETWORK_BACKEND
```

Comment:
i'm having trouble following this;
is it ok to remove `#ifdef WITH_NETWORK_BACKEND` ?
is `#ifdef HAVE_DNS_SD` dependent on `#ifdef WITH_NETWORK_BACKEND`?

Reply:
yes - `HAVE_DNS_SD` can only be set if `WITH_NETWORK_BACKEND` is set, and it is more appropriate for this code path.

Reply:
`HAVE_DNS_SD` is what allows empty strings to return as IP numbers - without that - calls to the networking will just fail with empty strings.

Reply:
[CMakeLists.txt](https://github.com/analogdevicesinc/libiio/blob/master/CMakeLists.txt#L235-L286)

`HAVE_DNS_SD` is set inside the `if(WITH_NETWORK_BACKEND)`